src/api/app/models/workflow/step/branch_package_step.rb

```diff
@@ -64,4 +66,15 @@ def create_branched_package
 
     target_package
   end
+
+  def write_report
+    return unless workflow_run = WorkflowRun.find(workflow_run_id)
+
+    # step_key = :branch_package
+    step_key = Workflow::SUPPORTED_STEPS.key(self.class).to_s
+    report = workflow_run.report ? JSON.parse(workflow_run.report) : {}
+    report[step_key] ||= []
+    report[step_key] << { 'source_project' => source_project_name, 'source_package' => source_package_name, 'target_project' => target_project_name, 'target_package' => target_package_name }
+    workflow_run.update(report: report.to_json)
```

Comment:
I would change `report` to `artifacts`

Reply:
Yes, why not. Let me finish everything using "report" and then I'll replace all the code.